lib/Frontend/FrontendOptions.cpp

```diff
@@ -116,14 +116,18 @@ void FrontendOptions::forAllOutputPaths(
     else
       fn(input.outputFilename());
   }
-  const std::string *outputs[] = {
-      &InputsAndOutputs.supplementaryOutputs().ModuleOutputPath,
-      &InputsAndOutputs.supplementaryOutputs().ModuleDocOutputPath,
-      &InputsAndOutputs.supplementaryOutputs().ObjCHeaderOutputPath};
-  for (const std::string *next : outputs) {
-    if (!next->empty())
-      fn(*next);
-  }
+  InputsAndOutputs.forEachInputProducingSupplementaryOutput(
+      [&](const InputFile &inp) -> void {
+        const SupplementaryOutputPaths outs =
+            inp.getPrimarySpecificPaths().SupplementaryOutputs;
+        const std::string *outputs[] = {&outs.ModuleOutputPath,
+                                        &outs.ModuleDocOutputPath,
+                                        &outs.ObjCHeaderOutputPath};
+        for (const std::string *next : outputs) {
+          if (!next->empty())
+            fn(*next);
```

Comment:
This whole function is bugging me—why just *these* three supplementary outputs? is it a guess at which ones are more than just build artifacts?—but I guess it doesn't have to be fixed in this PR.

Reply:
I agree, but that's what it was.

Reply:
Yep, not blaming you. I bet it was me in the first place.

Reply:
Thanks!